include/numerics/dense_matrix_impl.h

```diff
@@ -808,6 +793,34 @@ void DenseMatrix<T>::svd (DenseVector<Real> & sigma,
   _svd_lapack(sigma, U, VT);
 }
 
+template<typename T>
+void DenseMatrix<T>::svdinv (DenseMatrix<Number> & Inverse)
+
+{
+
+  // We use the LAPACK svd implementation
+  DenseMatrix<Number> U;
+  DenseVector<Real> sigma;
+  DenseMatrix<Number> VT;
+  _svd_lapack(sigma, U, VT);
+
+Inverse.resize(VT.n(),U.m());
+
+for (unsigned i =0; i<VT.n(); ++i)
+{
+  for (unsigned j = 0; j<U.m(); ++j)
+  {
+    for (unsigned k = 0; k<VT.m(); ++k)
+    {
+      Inverse(i,j) += VT(k,i)*U(j,k)/sigma(k);
```

Comment:
This isn't right for the case where some of the singular values are nearly zero, is it?

You don't want to divide by `sigma(k)` in that case in order to avoid introducing round-off error. See also the `_svd_solve_lapack` function where the user supplies an `rcond` value...